src/CarbonAware/src/Extensions/EmissionsDataExtensions.cs

```diff
@@ -0,0 +1,85 @@
+namespace CarbonAware.Extensions;
+
+public static class EmissionsDataExtensions
+{
+    public static IEnumerable<EmissionsData> RollingAverage(this IEnumerable<EmissionsData> data, TimeSpan windowSize = default, TimeSpan tickSize = default)
```

Comment:
It would be good to know why this is needed first of all (why the data that was provided by the backend api is not enough).